src/mongocxx/collection.hpp

```diff
@@ -602,57 +562,51 @@ class MONGOCXX_API collection {
    private:
     friend class database;
 
-    MONGOCXX_PRIVATE collection(const database& database, bsoncxx::stdx::string_view collection_name);
+    MONGOCXX_PRIVATE collection(const database& database,
+                                bsoncxx::stdx::string_view collection_name);
+
+    MONGOCXX_PRIVATE collection(const database& database,
+                                bsoncxx::stdx::string_view collection_name,
+                                mongoc_collection_t* collection);
```

Comment:
make this a void*. Then in the CPP static_cast it to mongoc_collection_t*. This is what we do elsewhere when we need to erase the types of mongoc structs in mongocxx headers.